fetch.js

```diff
@@ -66,21 +75,31 @@ function GetFileFromUrl(inputUrl, processFileCallback) {
         res.setEncoding("utf8");
         res.on("data", function (chunk) { rawResponse += chunk; });
         res.on("end", function () {
-            console.log("<- Finished reading: " + inputUrl);
-            processFileCallback(rawResponse);
+            console.log("<- Finished reading " + rawResponse.length + " bytes: " + inputUrl);
+            if (!processFileCallback(rawResponse))
+                console.log("  ***  Failed to GetFileFromUrl: " + inputUrl);
         });
     });
 }
 
-function GetApiFile(inputUrl, outputFilename) {
+function GetApiFile(inputUrl, outputFilename, retry = 0) {
+    if (retry == 10)
+        throw new Exception("  !!!!!!!!!!  Aborting GetApiFile: " + inputUrl);
```

Comment:
>  !!!!!!!!!!  Aborting GetApiFile:  [](start = 29, length = 35)

It may be a little more useful to say something like "Aborting after multiple/10 retries in GetApiFile: " + inputUrl